packages/opentelemetry-instrumentation-langchain/tests/test_lcel.py

```diff
@@ -270,44 +255,37 @@ def test_custom_llm(exporter):
 
 @pytest.mark.vcr
 def test_openai(exporter):
+    prompt = ChatPromptTemplate.from_messages(
+        [("system", "You are a helpful assistant"), ("human", "{input}")]
+    )
     model = ChatOpenAI(model="gpt-3.5-turbo")
+    chain = prompt | model
 
-    response = model.generate(
-        messages=[
-            [
-                SystemMessage(content="You are a helpful assistant"),
-                HumanMessage(content="Tell me a joke about OpenTelemetry"),
-            ]
-        ]
-    )
+    response = chain.invoke({"input": "Tell me a joke about OpenTelemetry"})
 
     spans = exporter.get_finished_spans()
 
     assert [
+        "ChatPromptTemplate.langchain.task",
         "openai.chat",
         "ChatOpenAI.langchain.task",
+        "RunnableSequence.langchain.workflow",
     ] == [span.name for span in spans]
 
     openai_span = next(
         span for span in spans if span.name == "ChatOpenAI.langchain.task"
     )
+    workflow_span = next(
+        span for span in spans if span.name == "RunnableSequence.langchain.workflow"
+    )
 
     assert openai_span.attributes[SpanAttributes.LLM_REQUEST_TYPE] == "chat"
     assert openai_span.attributes[SpanAttributes.LLM_REQUEST_MODEL] == "gpt-3.5-turbo"
-    assert (
```

Comment:
Did you remove these because they fail now? Cause we still expect the OpenAI span to be there (as it reports the token usage etc - although it might have been better to rely on the langchain instruments to on to provide that, but that's probably out of scope for now).

Reply:
TLDR: yes, they would fail because I left this from the new `callback_wrapper`. However, I have similar asserts on the input and output.

See an earlier comment regarding this: "I would recommend removing `_set_chat_request`, because all the relevant information is available in the other attributes. I provided this to be more inline with the past implementation, e.g. custom_chat_wrapper/_handle_request."

In `custom_chat_wrapper` there are 2 functions which populate these kinds of fields: `_handle_request` and `_handle_reponse`. I did not replicate the `should_send_prompts` parts because the output from the models is model-specific, furthermore, all the information can be found in `SpanAttributes.TRACELOOP_ENTITY_INPUT` and `SpanAttributes.TRACELOOP_ENTITY_OUTPUT`. One could write a parser for the callback arguments inputs / outputs / messages, but this could end up being a if-else spaghetti :)

Let me know please if you would like to have this parser populating the span attributes like `{SpanAttributes.LLM_PROMPTS}.{idx}.role`.

Reply:
But @tibor-reiss they should be generated from the OpenAI instrumentation, so they should always be there, no matter what you change in the LangChain instrumentation

Reply:
Yes, they are there. And they are passed into TRACELOOP_ENTITY_INPUT and TRACELOOP_ENTITY_OUTPUT (see the callbacks, e.g. on_chain_start / inputs and on_chain_end / outputs; they are just not parsed further into LLM_PROMPTS.*, yet.

Reply:
Not sure I follow @tibor-reiss - what happens if you re add those assertions? I expect the instrumentation for OpenAI to specify everything needed by these assertions

Reply:
Those are indeed provided by OpenAI, but as part of the callbacks, see `inputs`, `outputs` and messages `arguments`.

If I add back the deleted assertions the tests fail. The attributes starting with LLM_PROMPTS were previously (code is still there) filled in custom_chat_wrapper / _handle_request. This is what I call the "parser", and I can add it to callback_wrapper if you would like to (some basic version is already in `_set_chat_request`).

However, here is the `outputs` argument from `on_chain_end` for `test_openai`:
AIMessage(content="Why did OpenTelemetry break up with its significant other? Because it couldn't handle the constant tracing and monitoring of their relationship!", response_metadata={'token_usage': {'completion_tokens': 26, 'prompt_tokens': 24, 'total_tokens': 50}, 'model_name': 'gpt-3.5-turbo', 'system_fingerprint': 'fp_d9767fc5b9', 'finish_reason': 'stop', 'logprobs': None}, id='run-18c68937-8f10-4385-9cd5-1e279a0529c6-0', usage_metadata={'input_tokens': 24, 'output_tokens': 26, 'total_tokens': 50})

And here is the `outputs` from test_anthropic:
AIMessage(content="Why can't a bicycle stand up by itself? Because it's two-tired!", response_metadata={'id': 'msg_017fMG9SRDFTBhcD1ibtN1nK', 'model': 'claude-2.1', 'stop_reason': 'end_turn', 'stop_sequence': None, 'usage': {'input_tokens': 19, 'output_tokens': 22}}, id='run-4e13a0a2-c199-44ac-b0ab-b807ed023c62-0', usage_metadata={'input_tokens': 19, 'output_tokens': 22, 'total_tokens': 41})

They are similar, but there are still differences, e.g. one striking one is `model` vs `model_name`.

So my fear is that the parser we would write, it would just cause a lot of maintenance burden with newer versions. At the same time, all the inputs/messages/outputs are already provided in the SpanAttributes (in TRACELOOP_ENTITY_INPUT and TRACELOOP_ENTITY_OUTPUT), so the parser would be just a pretty printer :)

I don't have a strong preference though, I can implement it if you would like to.